### DISCUSSION TITLE
errors on fresh install on mac

### DISCUSSION THREAD

**Discussion by lgwjames** [lllyasviel/Fooocus#3321] [Q&A]
followed guide as posted and get errors
stderr: ERROR: Ignored the following versions that require a different python version: 1.6.2 Requires-Python >=3.7,<3.10; 1.6.3 Requires-Python >=3.7,<3.10; 1.7.0 Requires-Python >=3.7,<3.10; 1.7.1 Requires-Python >=3.7,<3.10
ERROR: Could not find a version that satisfies the requirement onnxruntime==1.18.1 (from versions: 1.12.0, 1.12.1, 1.13.1, 1.14.1, 1.15.0, 1.15.1, 1.16.0, 1.16.1, 1.16.2, 1.16.3)
ERROR: No matching distribution found for onnxruntime==1.18.1

CMD Failed requirements: install -r "requirements_versions.txt"
Traceback (most recent call last):
  File "/Users/Quad-Core/Fooocus/entry_with_update.py", line 46, in <module>
    from launch import *
  File "/Users/Quad-Core/Fooocus/launch.py", line 88, in <module>
    from modules.hash_cache import init_cache
  File "/Users/Quad-Core/Fooocus/modules/hash_cache.py", line 7, in <module>
    from modules.util import sha256, HASH_SHA256_LENGTH, get_file_from_folder_list
  File "/Users/Quad-Core/Fooocus/modules/util.py", line 8, in <module>
    import cv2
ModuleNotFoundError: No module named 'cv2'
(fooocus) Quad-Core@iMac fooocus % 

ideas?
i thought it would be the python 3.7 but tried and tried to downgrade and nothing

**Comment by mashb1t** [comment]
Please use Python 3.10 and try installing the requirements again.
From https://github.com/microsoft/onnxruntime/releases/tag/v1.18.0:
> Python version >=3.8 is now required for build.bat/build.sh (previously >=3.7). Note: If you have Python version <3.8, you can bypass the tools and use CMake directly.

**Reply by lgwjames** [reply]
(base) Quad-Core@iMac fooocus % python --version
Python 3.10.14
(base) Quad-Core@iMac fooocus % python entry_with_update.py
(base) Quad-Core@iMac fooocus % python entry_with_update.py
Update failed.
No module named 'pygit2'
Update succeeded.
[System ARGV] ['entry_with_update.py']
Python 3.10.14 (main, May  6 2024, 14:47:20) [Clang 14.0.6 ]
Fooocus version: 2.5.0
Error checking version for torchsde: No package metadata was found for torchsde
Installing requirements
Command: "/Users/Quad-Core/miniconda3/bin/python" -m pip install -r "requirements_versions.txt" --prefer-binary
Error code: 1
stdout: Collecting torchsde==0.2.6 (from -r requirements_versions.txt (line 1))
  Using cached torchsde-0.2.6-py3-none-any.whl.metadata (5.3 kB)
Collecting einops==0.8.0 (from -r requirements_versions.txt (line 2))
  Using cached einops-0.8.0-py3-none-any.whl.metadata (12 kB)
Collecting transformers==4.42.4 (from -r requirements_versions.txt (line 3))
  Using cached transformers-4.42.4-py3-none-any.whl.metadata (43 kB)
Collecting safetensors==0.4.3 (from -r requirements_versions.txt (line 4))
  Using cached safetensors-0.4.3-cp310-cp310-macosx_10_12_x86_64.whl.metadata (3.8 kB)
Collecting accelerate==0.32.1 (from -r requirements_versions.txt (line 5))
  Using cached accelerate-0.32.1-py3-none-any.whl.metadata (18 kB)
Requirement already satisfied: pyyaml==6.0.1 in /Users/Quad-Core/miniconda3/lib/python3.10/site-packages (from -r requirements_versions.txt (line 6)) (6.0.1)
Requirement already satisfied: pillow==10.4.0 in /Users/Quad-Core/miniconda3/lib/python3.10/site-packages (from -r requirements_versions.txt (line 7)) (10.4.0)
Collecting scipy==1.14.0 (from -r requirements_versions.txt (line 8))
  Using cached scipy-1.14.0-cp310-cp310-macosx_10_9_x86_64.whl.metadata (60 kB)
Requirement already satisfied: tqdm==4.66.4 in /Users/Quad-Core/miniconda3/lib/python3.10/site-packages (from -r requirements_versions.txt (line 9)) (4.66.4)
Collecting psutil==6.0.0 (from -r requirements_versions.txt (line 10))
  Using cached psutil-6.0.0-cp36-abi3-macosx_10_9_x86_64.whl.metadata (21 kB)
Collecting pytorch_lightning==2.3.3 (from -r requirements_versions.txt (line 11))
  Using cached pytorch_lightning-2.3.3-py3-none-any.whl.metadata (21 kB)
Collecting omegaconf==2.3.0 (from -r requirements_versions.txt (line 12))
  Using cached omegaconf-2.3.0-py3-none-any.whl.metadata (3.9 kB)
Collecting gradio==3.41.2 (from -r requirements_versions.txt (line 13))
  Using cached gradio-3.41.2-py3-none-any.whl.metadata (17 kB)
Collecting pygit2==1.15.1 (from -r requirements_versions.txt (line 14))
  Using cached pygit2-1.15.1-cp310-cp310-macosx_10_9_universal2.whl.metadata (3.3 kB)
Collecting opencv-contrib-python-headless==4.10.0.84 (from -r requirements_versions.txt (line 15))
  Using cached opencv-contrib-python-headless-4.10.0.84.tar.gz (150.5 MB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'done'
  Preparing metadata (pyproject.toml): started
  Preparing metadata (pyproject.toml): finished with status 'done'
Collecting httpx==0.27.0 (from -r requirements_versions.txt (line 16))
  Using cached httpx-0.27.0-py3-none-any.whl.metadata (7.2 kB)

stderr: ERROR: Ignored the following versions that require a different python version: 1.6.2 Requires-Python >=3.7,<3.10; 1.6.3 Requires-Python >=3.7,<3.10; 1.7.0 Requires-Python >=3.7,<3.10; 1.7.1 Requires-Python >=3.7,<3.10
ERROR: Could not find a version that satisfies the requirement onnxruntime==1.18.1 (from versions: 1.12.0, 1.12.1, 1.13.1, 1.14.1, 1.15.0, 1.15.1, 1.16.0, 1.16.1, 1.16.2, 1.16.3)
ERROR: No matching distribution found for onnxruntime==1.18.1

CMD Failed requirements: install -r "requirements_versions.txt"
Traceback (most recent call last):
  File "/Users/Quad-Core/Fooocus/entry_with_update.py", line 46, in <module>
    from launch import *
  File "/Users/Quad-Core/Fooocus/launch.py", line 88, in <module>
    from modules.hash_cache import init_cache
  File "/Users/Quad-Core/Fooocus/modules/hash_cache.py", line 7, in <module>
    from modules.util import sha256, HASH_SHA256_LENGTH, get_file_from_folder_list
  File "/Users/Quad-Core/Fooocus/modules/util.py", line 8, in <module>
    import cv2
ModuleNotFoundError: No module named 'cv2'
(base) Quad-Core@iMac fooocus % (base) Quad-Core@iMac fooocus % python entry_with_update.py

**Reply by lgwjames** [reply]
(base) Quad-Core@iMac fooocus % git clone https://github.com/microsoft/onnxruntime.git
Cloning into 'onnxruntime'...
remote: Enumerating objects: 391773, done.
remote: Counting objects: 100% (4216/4216), done.
remote: Compressing objects: 100% (2093/2093), done.
Receiving objects:  10% (42388/391773), 148.04 MiB | 8.63 MiB/s

**Comment by lgwjames** [comment]
looks like this  pip install -r requirements_versions.txt is saying it is doing something, when its not, as for last hour, see that they do install when i do them manually and then run again something missing

**Comment by lgwjames** [comment]
(fooocus) Quad-Core@iMac fooocus % pip install -r requirements_versions.txt
Requirement already satisfied: torchsde==0.2.6 in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from -r requirements_versions.txt (line 1)) (0.2.6)
Requirement already satisfied: einops==0.8.0 in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from -r requirements_versions.txt (line 2)) (0.8.0)
Requirement already satisfied: transformers==4.42.4 in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from -r requirements_versions.txt (line 3)) (4.42.4)
Requirement already satisfied: safetensors==0.4.3 in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from -r requirements_versions.txt (line 4)) (0.4.3)
Collecting accelerate==0.32.1
  Using cached accelerate-0.32.1-py3-none-any.whl (314 kB)
Requirement already satisfied: pyyaml==6.0.1 in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from -r requirements_versions.txt (line 6)) (6.0.1)
Requirement already satisfied: pillow==10.4.0 in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from -r requirements_versions.txt (line 7)) (10.4.0)
Requirement already satisfied: scipy==1.14.0 in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from -r requirements_versions.txt (line 8)) (1.14.0)
Requirement already satisfied: tqdm==4.66.4 in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from -r requirements_versions.txt (line 9)) (4.66.4)
Collecting psutil==6.0.0
  Using cached psutil-6.0.0-cp36-abi3-macosx_10_9_x86_64.whl (250 kB)
Collecting pytorch_lightning==2.3.3
  Using cached pytorch_lightning-2.3.3-py3-none-any.whl (812 kB)
Collecting omegaconf==2.3.0
  Using cached omegaconf-2.3.0-py3-none-any.whl (79 kB)
Collecting gradio==3.41.2
  Using cached gradio-3.41.2-py3-none-any.whl (20.1 MB)
Collecting pygit2==1.15.1
  Using cached pygit2-1.15.1-cp310-cp310-macosx_10_9_universal2.whl (5.9 MB)
Collecting opencv-contrib-python-headless==4.10.0.84
  Using cached opencv-contrib-python-headless-4.10.0.84.tar.gz (150.5 MB)
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Collecting httpx==0.27.0
  Using cached httpx-0.27.0-py3-none-any.whl (75 kB)
ERROR: Could not find a version that satisfies the requirement onnxruntime==1.18.1 (from versions: 1.12.0, 1.12.1, 1.13.1, 1.14.1, 1.15.0, 1.15.1, 1.16.0, 1.16.1, 1.16.2, 1.16.3)
ERROR: No matching distribution found for onnxruntime==1.18.1
(fooocus) Quad-Core@iMac fooocus %

**Reply by mashb1t** [reply]
This is the main issue, which also is most likely blocking the automated update of Fooocus. Check what the reason is why the package only shows as compatible until 1.16.3 and not above.
`python -m pip check` might help here. also check out `python -m pip install onnxruntime== --dry-run --python-version 3.10 --no-deps --target foo` and https://stackoverflow.com/questions/66627014/how-to-know-what-python-version-a-package-is-compatible-with/74365779#74365779

**Comment by lgwjames** [comment]
(fooocus) Quad-Core@iMac fooocus % python entry_with_update.py             
Update failed.
No module named 'pygit2'
Update succeeded.
[System ARGV] ['entry_with_update.py']
Python 3.10.14 (main, May  6 2024, 14:47:20) [Clang 14.0.6 ]
Fooocus version: 2.5.0
Error checking version for accelerate: No package metadata was found for accelerate
Installing requirements

i bet if i go look for accelerate and install that error will go

**Comment by lgwjames** [comment]
yep, so errors in the install file

(fooocus) Quad-Core@iMac fooocus % pip install accelerate
Collecting accelerate
  Using cached accelerate-0.32.1-py3-none-any.whl (314 kB)
Collecting psutil
  Using cached psutil-6.0.0-cp36-abi3-macosx_10_9_x86_64.whl (250 kB)
Requirement already satisfied: huggingface-hub in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from accelerate) (0.24.0)
Requirement already satisfied: numpy<2.0.0,>=1.17 in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from accelerate) (1.26.4)
Requirement already satisfied: torch>=1.10.0 in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from accelerate) (2.1.0)
Requirement already satisfied: pyyaml in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from accelerate) (6.0.1)
Requirement already satisfied: safetensors>=0.3.1 in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from accelerate) (0.4.3)
Requirement already satisfied: packaging>=20.0 in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from accelerate) (24.1)
Requirement already satisfied: typing-extensions in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from torch>=1.10.0->accelerate) (4.12.2)
Requirement already satisfied: jinja2 in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from torch>=1.10.0->accelerate) (3.1.4)
Requirement already satisfied: filelock in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from torch>=1.10.0->accelerate) (3.15.4)
Requirement already satisfied: networkx in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from torch>=1.10.0->accelerate) (3.3)
Requirement already satisfied: sympy in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from torch>=1.10.0->accelerate) (1.13.1)
Requirement already satisfied: fsspec in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from torch>=1.10.0->accelerate) (2024.6.1)
Requirement already satisfied: tqdm>=4.42.1 in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from huggingface-hub->accelerate) (4.66.4)
Requirement already satisfied: requests in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from huggingface-hub->accelerate) (2.32.3)
Requirement already satisfied: MarkupSafe>=2.0 in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from jinja2->torch>=1.10.0->accelerate) (2.1.5)
Requirement already satisfied: charset-normalizer<4,>=2 in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from requests->huggingface-hub->accelerate) (3.3.2)
Requirement already satisfied: certifi>=2017.4.17 in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from requests->huggingface-hub->accelerate) (2024.7.4)
Requirement already satisfied: idna<4,>=2.5 in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from requests->huggingface-hub->accelerate) (3.7)
Requirement already satisfied: urllib3<3,>=1.21.1 in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from requests->huggingface-hub->accelerate) (2.2.2)
Requirement already satisfied: mpmath<1.4,>=1.1.0 in /Users/Quad-Core/miniconda3/envs/fooocus/lib/python3.10/site-packages (from sympy->torch>=1.10.0->accelerate) (1.3.0)
Installing collected packages: psutil, accelerate
Successfully installed accelerate-0.32.1 psutil-6.0.0
(fooocus) Quad-Core@iMac fooocus % python entry_with_update.py
Update failed.
No module named 'pygit2'
Update succeeded.
[System ARGV] ['entry_with_update.py']
Python 3.10.14 (main, May  6 2024, 14:47:20) [Clang 14.0.6 ]
Fooocus version: 2.5.0
Error checking version for pytorch_lightning: No package metadata was found for pytorch_lightning
Installing requirements

**Comment by lgwjames** [comment]
GIVEN UP!!!!

installed everything as needed and on the last install i got 

(base) Quad-Core@iMac fooocus % pip install torchsde
Collecting torchsde
  Using cached torchsde-0.2.6-py3-none-any.whl.metadata (5.3 kB)
Requirement already satisfied: numpy>=1.19 in /Users/Quad-Core/miniconda3/lib/python3.12/site-packages (from torchsde) (2.0.0)
Collecting scipy>=1.5 (from torchsde)
  Using cached scipy-1.14.0-cp312-cp312-macosx_10_9_x86_64.whl.metadata (60 kB)
Requirement already satisfied: torch>=1.6.0 in /Users/Quad-Core/miniconda3/lib/python3.12/site-packages (from torchsde) (2.2.2)
Collecting trampoline>=0.1.2 (from torchsde)
  Using cached trampoline-0.1.2-py3-none-any.whl.metadata (10 kB)
Requirement already satisfied: filelock in /Users/Quad-Core/miniconda3/lib/python3.12/site-packages (from torch>=1.6.0->torchsde) (3.15.4)
Requirement already satisfied: typing-extensions>=4.8.0 in /Users/Quad-Core/miniconda3/lib/python3.12/site-packages (from torch>=1.6.0->torchsde) (4.12.2)
Requirement already satisfied: sympy in /Users/Quad-Core/miniconda3/lib/python3.12/site-packages (from torch>=1.6.0->torchsde) (1.13.1)
Requirement already satisfied: networkx in /Users/Quad-Core/miniconda3/lib/python3.12/site-packages (from torch>=1.6.0->torchsde) (3.3)
Requirement already satisfied: jinja2 in /Users/Quad-Core/miniconda3/lib/python3.12/site-packages (from torch>=1.6.0->torchsde) (3.1.4)
Requirement already satisfied: fsspec in /Users/Quad-Core/miniconda3/lib/python3.12/site-packages (from torch>=1.6.0->torchsde) (2024.6.1)
Requirement already satisfied: MarkupSafe>=2.0 in /Users/Quad-Core/miniconda3/lib/python3.12/site-packages (from jinja2->torch>=1.6.0->torchsde) (2.1.5)
Requirement already satisfied: mpmath<1.4,>=1.1.0 in /Users/Quad-Core/miniconda3/lib/python3.12/site-packages (from sympy->torch>=1.6.0->torchsde) (1.3.0)
Downloading torchsde-0.2.6-py3-none-any.whl (61 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 61.2/61.2 kB 1.2 MB/s eta 0:00:00
Downloading scipy-1.14.0-cp312-cp312-macosx_10_9_x86_64.whl (39.2 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 39.2/39.2 MB 8.8 MB/s eta 0:00:00
Using cached trampoline-0.1.2-py3-none-any.whl (5.2 kB)
Installing collected packages: trampoline, scipy, torchsde
Successfully installed scipy-1.14.0 torchsde-0.2.6 trampoline-0.1.2
(base) Quad-Core@iMac fooocus % python entry_with_update.py
Already up-to-date
Update succeeded.
[System ARGV] ['entry_with_update.py']
Python 3.12.4 | packaged by Anaconda, Inc. | (main, Jun 18 2024, 10:14:12) [Clang 14.0.6 ]
Fooocus version: 2.5.0
Error checking version for einops: No package metadata was found for einops
Installing requirements
Couldn't install requirements.
Command: "/Users/Quad-Core/miniconda3/bin/python" -m pip install -r "requirements_versions.txt" --prefer-binary
Error code: 1
stdout: Requirement already satisfied: torchsde==0.2.6 in /Users/Quad-Core/miniconda3/lib/python3.12/site-packages (from -r requirements_versions.txt (line 1)) (0.2.6)
Collecting einops==0.8.0 (from -r requirements_versions.txt (line 2))
  Using cached einops-0.8.0-py3-none-any.whl.metadata (12 kB)
Collecting transformers==4.42.4 (from -r requirements_versions.txt (line 3))
  Using cached transformers-4.42.4-py3-none-any.whl.metadata (43 kB)
Collecting safetensors==0.4.3 (from -r requirements_versions.txt (line 4))
  Using cached safetensors-0.4.3-cp312-cp312-macosx_10_12_x86_64.whl.metadata (3.8 kB)
Collecting accelerate==0.32.1 (from -r requirements_versions.txt (line 5))
  Using cached accelerate-0.32.1-py3-none-any.whl.metadata (18 kB)
Collecting pyyaml==6.0.1 (from -r requirements_versions.txt (line 6))
  Using cached PyYAML-6.0.1-cp312-cp312-macosx_10_9_x86_64.whl.metadata (2.1 kB)
Requirement already satisfied: pillow==10.4.0 in /Users/Quad-Core/miniconda3/lib/python3.12/site-packages (from -r requirements_versions.txt (line 7)) (10.4.0)
Requirement already satisfied: scipy==1.14.0 in /Users/Quad-Core/miniconda3/lib/python3.12/site-packages (from -r requirements_versions.txt (line 8)) (1.14.0)
Requirement already satisfied: tqdm==4.66.4 in /Users/Quad-Core/miniconda3/lib/python3.12/site-packages (from -r requirements_versions.txt (line 9)) (4.66.4)
Collecting psutil==6.0.0 (from -r requirements_versions.txt (line 10))
  Using cached psutil-6.0.0-cp36-abi3-macosx_10_9_x86_64.whl.metadata (21 kB)
Collecting pytorch_lightning==2.3.3 (from -r requirements_versions.txt (line 11))
  Using cached pytorch_lightning-2.3.3-py3-none-any.whl.metadata (21 kB)
Collecting omegaconf==2.3.0 (from -r requirements_versions.txt (line 12))
  Using cached omegaconf-2.3.0-py3-none-any.whl.metadata (3.9 kB)
Collecting gradio==3.41.2 (from -r requirements_versions.txt (line 13))
  Using cached gradio-3.41.2-py3-none-any.whl.metadata (17 kB)
Requirement already satisfied: pygit2==1.15.1 in /Users/Quad-Core/miniconda3/lib/python3.12/site-packages (from -r requirements_versions.txt (line 14)) (1.15.1)
Collecting opencv-contrib-python-headless==4.10.0.84 (from -r requirements_versions.txt (line 15))
  Using cached opencv-contrib-python-headless-4.10.0.84.tar.gz (150.5 MB)
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'

stderr:   error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [33 lines of output]
      Traceback (most recent call last):
        File "/Users/Quad-Core/miniconda3/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/Users/Quad-Core/miniconda3/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/Quad-Core/miniconda3/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 112, in get_requires_for_build_wheel
          backend = _build_backend()
                    ^^^^^^^^^^^^^^^^
        File "/Users/Quad-Core/miniconda3/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 77, in _build_backend
          obj = import_module(mod_path)
                ^^^^^^^^^^^^^^^^^^^^^^^
        File "/Users/Quad-Core/miniconda3/lib/python3.12/importlib/__init__.py", line 90, in import_module
          return _bootstrap._gcd_import(name[level:], package, level)
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
        File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
        File "<frozen importlib._bootstrap>", line 1310, in _find_and_load_unlocked
        File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
        File "<frozen importlib._bootstrap>", line 1387, in _gcd_import
        File "<frozen importlib._bootstrap>", line 1360, in _find_and_load
        File "<frozen importlib._bootstrap>", line 1331, in _find_and_load_unlocked
        File "<frozen importlib._bootstrap>", line 935, in _load_unlocked
        File "<frozen importlib._bootstrap_external>", line 995, in exec_module
        File "<frozen importlib._bootstrap>", line 488, in _call_with_frames_removed
        File "/private/var/folders/6l/xrkxgs657g7_vq50479x7lc40000gp/T/pip-build-env-v_2_mdia/overlay/lib/python3.12/site-packages/setuptools/__init__.py", line 16, in <module>
          import setuptools.version
        File "/private/var/folders/6l/xrkxgs657g7_vq50479x7lc40000gp/T/pip-build-env-v_2_mdia/overlay/lib/python3.12/site-packages/setuptools/version.py", line 1, in <module>
          import pkg_resources
        File "/private/var/folders/6l/xrkxgs657g7_vq50479x7lc40000gp/T/pip-build-env-v_2_mdia/overlay/lib/python3.12/site-packages/pkg_resources/__init__.py", line 2172, in <module>
          register_finder(pkgutil.ImpImporter, find_on_path)
                          ^^^^^^^^^^^^^^^^^^^
      AttributeError: module 'pkgutil' has no attribute 'ImpImporter'. Did you mean: 'zipimporter'?
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

CMD Failed requirements: install -r "requirements_versions.txt"
Traceback (most recent call last):
  File "/Users/Quad-Core/Fooocus/entry_with_update.py", line 46, in <module>
    from launch import *
  File "/Users/Quad-Core/Fooocus/launch.py", line 88, in <module>
    from modules.hash_cache import init_cache
  File "/Users/Quad-Core/Fooocus/modules/hash_cache.py", line 7, in <module>
    from modules.util import sha256, HASH_SHA256_LENGTH, get_file_from_folder_list
  File "/Users/Quad-Core/Fooocus/modules/util.py", line 8, in <module>
    import cv2
ModuleNotFoundError: No module named 'cv2'
(base) Quad-Core@iMac fooocus % 


and removed everything i just spent a hour installing !

**Reply by mashb1t** [reply]
Fooocus isn't compatible with Python 3.12, why did you switch to it?
There are ways to make it compatible, but you have to do a few things, which you can find in https://github.com/lllyasviel/Fooocus/issues/2938 and https://github.com/lllyasviel/Fooocus/pull/2927.

**Comment by lgwjames** [comment]
still the same, everything installed , even cloned 1.18.1 and errors, errors, errors

**Comment by lgwjames** [comment]
so i just did git checkout 2.3.0. and then pip install -r requirements_versions.txt
loads off installs happened. which hasn't happened for 2 days.

downloading default files

**Comment by lgwjames** [comment]
ok making images, some still black, 
also my config file not working correctly?
as its not pre loading my loras
